gnat2goto/driver/tree_walk.adb

```diff
@@ -337,6 +341,11 @@ package body Tree_Walk is
    with Pre => Nkind (N) in N_Subprogram_Specification;
    --  Insert the subprogram specification into the symbol table
 
+   procedure Register_Type_Declaration (N : Node_Id; E : Entity_Id)
+   with Pre => Nkind (N) = N_Full_Type_Declaration;
```

Comment:
Is there something we could assert about `E` as a precondition. It is not necessary but would be nice as documentation.

Reply:
We could have "and Is_Type (E)" added to the precondition.